test/test_xla_sharding.py

```diff
@@ -55,6 +54,21 @@ def test_mark_sharding(self):
     actual = (xt1 @ xt2.T).cpu()
     self.assertTrue(torch.allclose(expected, actual))
 
+  def test_mark_sharding_4d(self):
+    t = torch.randn(2, 4, 8, 16, device='cpu')
+    expected = t + t
+
+    xt = t.to(xm.xla_device())
+    n_devices = xm.xrt_world_size()
```

Comment:
Does CI run this test or we only run it on TPU?

Reply:
We only run cpp tests -- covers the internal changes that affects the non-spmd code paths -- and the python API tests are disabled [link](https://github.com/pytorch/xla/blob/e0dd3e08fe393fc64f769a1fc30861e7e793b019/test/run_tests.sh#L130). I will re-enable it after debugging/ adding the API unit tests.